dark/utils.py

```diff
@@ -360,3 +361,69 @@ def openOr(filename, mode="r", defaultFp=None, specialCaseHyphen=True):
     else:
         with open(filename, mode) as fp:
             yield fp
+
+
+def intsToRanges(numbers: Iterable[int]) -> list[tuple[int, int]]:
+    """
+    Convert a set of integers into a list of consecutive integer ranges.
```

Comment:
When I read this I thought it did something different than what it does. I would call the new representation a list of intervals, I think range suggests something that can be iterated over in the context of python.
Also I would say that it changes the representation of the set of integers instead of converting them. 
> Given a set of integers, `A`, this function finds a set of intervals `[a_i, b_i]` (for `i` in some finite index set), such that `A` is the union of `[a_i, b_i]` over all `i` and returns a list of ordered-pairs `(a_i, b_i)`.

Reply:
Done, thanks! And I inserted "mutually disjoint" before "a set of intervals" :-)